tests/expressions_window/tests.py

```diff
@@ -0,0 +1,710 @@
+            order_by='department')
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Williams', 37000, 'Accounting', 60000),
+            ('Jenson', 45000, 'Accounting', 60000),
+            ('Adams', 50000, 'Accounting', 60000),
+            ('Jones', 45000, 'Accounting', 60000),
+            ('Wilkinson', 60000, 'IT', 60000),
+            ('Moore', 34000, 'IT', 60000),
+            ('Johnson', 80000, 'Management', 60000),
+            ('Johnson', 40000, 'Marketing', 60000),
+            ('Miller', 100000, 'Management', 60000),
+            ('Smith', 38000, 'Marketing', 60000),
+            ('Brown', 53000, 'Sales', 60000),
+            ('Smith', 55000, 'Sales', 60000),
+        ], transform=lambda row: (row.name, row.salary, row.department, row.lead_default), ordered=False)
+
+    def test_ntile(self):
+        qs = WindowTestModel.objects.annotate(ntile=Window(
+            expression=Ntile(num_buckets=4),
+            order_by='-salary'),
+        ).order_by('ntile', '-salary', 'name')
+
+        self.assertQuerysetEqual(qs, [
+            ('Miller', 'Management', 100000, 1),
+            ('Johnson', 'Management', 80000, 1),
+            ('Wilkinson', 'IT', 60000, 1),
+            ('Smith', 'Sales', 55000, 2),
+            ('Brown', 'Sales', 53000, 2),
+            ('Adams', 'Accounting', 50000, 2),
+            ('Jenson', 'Accounting', 45000, 3),
+            ('Jones', 'Accounting', 45000, 3),
+            ('Johnson', 'Marketing', 40000, 3),
+            ('Smith', 'Marketing', 38000, 4),
+            ('Williams', 'Accounting', 37000, 4),
+            ('Moore', 'IT', 34000, 4),
+        ], lambda x: (x.name, x.department, x.salary, x.ntile),)
+
+    def test_percent_rank(self):
+        qs = WindowTestModel.objects.annotate(percent_rank=Window(
+            expression=PercentRank(),
+            order_by=ExpressionList(OrderBy(F('salary')), OrderBy(F('name')))),
+        ).order_by('percent_rank')
+
+        # round for test unification, Oracle and PostgreSQL have a different number
+        # of decimal places
+        self.assertQuerysetEqual(qs, [
+            ('Moore', 'IT', 34000, 0.0),
+            ('Williams', 'Accounting', 37000, 0.090909090909091),
+            ('Smith', 'Marketing', 38000, 0.181818181818182),
+            ('Johnson', 'Marketing', 40000, 0.272727272727273),
+            ('Jenson', 'Accounting', 45000, 0.363636363636364),
+            ('Jones', 'Accounting', 45000, 0.454545454545455),
+            ('Adams', 'Accounting', 50000, 0.545454545454545),
+            ('Brown', 'Sales', 53000, 0.636363636363636),
+            ('Smith', 'Sales', 55000, 0.727272727272727),
+            ('Wilkinson', 'IT', 60000, 0.818181818181818),
+            ('Johnson', 'Management', 80000, 0.909090909090909),
+            ('Miller', 'Management', 100000, 1.0),
+        ], transform=lambda row: (row.name, row.department, row.salary, round(row.percent_rank, 15)),)
+
+    def test_nth_returns_null(self):
```

Comment:
This test uses an integer-field, but the name may be misleading because of this empty string/null. I've added a remark about the empty strings/null for `nth_value` in the documentation; I don't want to trigger a build just for this commit. For `lead` and `lag`, the default null is passed explicitly in the constructor.